### ISSUE TITLE
[Security] Storing password as plain text

### ISSUE THREAD

**Issue by BaranMichal25** [parse-community/Parse-SDK-Flutter#444]
I just realised that the library is saving ParseUser data on most of successful [user operations](
https://github.com/parse-community/Parse-SDK-Flutter/blob/master/lib/src/objects/parse_user.dart#L430), which might seem handy, but the problem is that it is stored as [plain text](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/lib/src/objects/parse_base.dart#L181), **including password**, which is definitely a security issue.

I believe we should change library's behaviour here or at least make it more clear that the library is storing user passwords as plain text.
What do you think?

Or am I missing something? I know that encryption can be turned on in CoreStore library, but right now that would be suboptimal solution, because all data would be automatically encrypted.

**Comment by RodrigoSMarques** [comment]
Hi @BaranMichal25 

Using Sembast, encryption is enabled by default, and you can enter an encryption password.
```dart
     ...
      coreStore: await CoreStoreSembastImp.getInstance(password: 'password'),
     ...
```

**Comment by BaranMichal25** [comment]
Yes, but then all data will be encrypted, right?
In which case, also loading all other classes and their data would be needlessly slow

**Comment by RodrigoSMarques** [comment]
Yes, all data will be encrypted.

Is it really slow? How much data will you store locally?
Sembast is a pure dart database and very fast in performance tests.

**Comment by fischerscode** [comment]
CoreStoreSembastImp is always encrypted (except on web), the password has a [default value](https://github.com/parse-community/Parse-SDK-Flutter/blob/2088bf1b07594a080b91d5002eda0515dae63568/lib/src/storage/core_store_sp_impl.dart#L9).

I think, the password should not be stored in any database neither in clear text, nor encrypted. Only storing a hashed password is acceptable.

When inspecting my saved user object (`ParseUser.currentUser()`), it indeed still contains the password.

I have three solutions to solve this problem:
1. provide the password when calling login()
2. storing the password in a field (instead of `_objectData`) (should be the easiest to implement)
3. removing the password at the `saveInStorage` call

I think the second one is the best. I will create a branch to test this change, soon.

**Comment by fischerscode** [comment]
I have implemented option two in [this branch](https://github.com/fischerscode/Parse-SDK-Flutter/tree/issue/%23444).
Testing is highly appreciated as I do not know which side effects this change has.

**Comment by fischerscode** [comment]
> I have implemented option two in [this branch](https://github.com/fischerscode/Parse-SDK-Flutter/tree/issue/%23444).


I tested this change in my project and I didn't encounter any problems.

**Comment by fischerscode** [comment]
Option 2 was merged with #446 into release/1.0.28.

**Comment by BaranMichal25** [comment]
Nice, thanks for taking care of it :)

**Comment by BaranMichal25** [comment]
Actually, can you please explain how this works?
I would think that password is part of [objectData](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/lib/src/objects/parse_base.dart#L11), so we're still [storing](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/lib/src/objects/parse_base.dart#L180) it in the database? Or is it hashed at this point?

**Comment by fischerscode** [comment]
The password is now stored [here](https://github.com/parse-community/Parse-SDK-Flutter/blob/b9f51a644a240374061856bdca78bbcb5ddce01b/packages/dart/lib/src/objects/parse_user.dart#L42) and this variable never gets stored in the `CoreStore`.
(Note: #446 was merged into release/1.0.28. Master does currently not have this change.)

**Comment by BaranMichal25** [comment]
All righty, thanks again :)
